File: 3016/solution.ts
function minimumPushes(word: string): number {
  const counts = Array.from({ length: 26 }, () => 0);
  for (let i = 0; i < word.length; i++) {
    counts[word.charCodeAt(i) - "a".charCodeAt(0)]++;
  }
  counts.sort((a, b) => b - a);
  let res = 0;
  for (let i = 0; i <= 25; i++) {
    res += Math.ceil((i + 1) / 8) * counts[i];
  }
  return res;
}

File: 3016/solution.java
import java.util.Arrays;

class Solution {
    public int minimumPushes(String word) {
        int[] counts = new int[26];
        for (int i = 0; i < word.length(); i++) {
            counts[word.charAt(i) - 'a']++;
        }
        Arrays.sort(counts);
        int res = 0;
        for (int i = 25; i >= 0; i--) {
            res += Math.ceil((25 - i + 1) / 8.0) * counts[i];
        }
        return res;
    }
}

// time O(n)
// space O(1)